src/cli/commands/telemetry/telemetry.spec.ts

```diff
@@ -0,0 +1,94 @@
+import fs from "fs";
+import mockFs from "mock-fs";
+import os from "os";
+import path from "path";
+import { logger } from "../../../core";
+import { ENV_FILENAME } from "../../../core/constants";
+import { telemetry } from "./telemetry";
+
+jest.mock("../../../core/utils/logger", () => {
+  return {
+    logger: {
+      error: jest.fn(),
+      log: jest.fn(),
+      warn: jest.fn(),
+      silly: jest.fn(),
+    },
+    logGiHubIssueMessageAndExit: jest.fn(),
+  };
+});
+
+describe("swa telemetry", () => {
+  afterEach(() => {
+    mockFs.restore();
+  });
+
+  it("Telemetry should be enabled by default", async () => {
+    mockFs();
+    await telemetry({ status: true });
+    expect(logger.log).toHaveBeenNthCalledWith(1, "Telemetry capturing is enabled.");
+  });
+
+  it("should store disable telemetry setting in .env file", async () => {
+    await telemetry({ disable: true });
+    const envFile = path.join(os.homedir(), ".swa", ENV_FILENAME);
+    const envFileContent = fs.readFileSync(envFile, "utf-8");
+
+    expect(envFileContent).toMatchSnapshot(`
+        "SWA_DISABLE_TELEMETRY=true"
+        `);
+  });
+
+  it("should log the telemetry setting as disabled", async () => {
+    await telemetry({ status: true });
+    expect(logger.log).toHaveBeenNthCalledWith(3, "Telemetry capturing is disabled.");
+  });
+
+  it("should store enable telemetry setting in .env file", async () => {
+    await telemetry({ enable: true });
+    const envFile = path.join(os.homedir(), ".swa", ENV_FILENAME);
+    const envFileContent = fs.readFileSync(envFile, "utf-8");
+
+    expect(envFileContent).toMatchSnapshot(`
+        "SWA_DISABLE_TELEMETRY=false"
+        `);
+  });
+
+  it("should log the telemetry setting as enabled", async () => {
+    await telemetry({ status: true });
+    expect(logger.log).toHaveBeenNthCalledWith(5, "Telemetry capturing is enabled.");
+  });
+
+  it("should disable telemetry when both --enable and --disable flags are used", async () => {
+    await telemetry({ disable: true, enable: true });
```

Comment:
we should call it as unsupported as we shouldn't allow passing those flags together.

Reply:
@Reshmi-Sriram thoughts on using --enable and --disable together?

Reply:
We should keep it simple.
just support passing only one option here:

--enable
--disable
--status

any combination of these will not make sense and in my opinion should not be allowed and appropriate message can be given to the customer.